### PR TITLE
Implement "Range Select" for Manga Chapters

### LINKED ISSUE #1493: Download Feature Request: Range Selection
**App version:**
0.7.4
**Issue/Request:**

[REQUEST]  Can we have the range selection to download manga chapters. 
Eg. Select starting chapter then select ending chapter?
Then all the chapters between those will be downloaded

**Other details:**

I used this feature in other manga app and is very convenient to have. 

screenshot_2018-06-17-15-27-19-666

### PR DESCRIPTION
**Pull request by Flat**
Closes #1493 and partially #1293

### CODE REVIEW

**Comment by BakaSai** [comment]
A much wanted feature of mine as well, I wonder why it hasn't been merged yet 🤔

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/chapter/ChaptersController.kt [hunk 1]
```diff
@@ -349,6 +350,27 @@ class ChaptersController : NucleusController<ChaptersPresenter>(),
         actionMode?.invalidate()
     }
 
+    private fun selectBetween() {
+        val adapter = adapter ?: return
+        if (selectedItems.size == 2) {
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/manga/chapter/ChaptersController.kt]: You could potentially enable this only when there's 2 items selected, in `onPrepareActionMode`.

Comment (hXtreme) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/chapter/ChaptersController.kt]: Seems like a nice idea, I could try to implement it but I'm not sure if I know how to do it.

My idea on how to implement it:  
1. Select Between option starts out as hidden.
2. Whenever a chapter was selected check to see if exactly 2 chapters are selected.
    a. If there are then set visibility to visible
    b. Else set it to hidden
    c. Finally do something to tell the UI to update.

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/chapter/ChaptersController.kt]: Sounds about right. I had it as a disabled option in my mind, although I'm not entirely sure if menu items can be set as enabled/disabled, or if that's the same as visible/hidden.

You _should_ be able to invalidate the menu as needed to get it to update.

Comment (hXtreme) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/chapter/ChaptersController.kt]: I do not know what this "invalidate the menu" means.

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/chapter/ChaptersController.kt]: https://developer.android.com/reference/android/app/Activity#invalidateOptionsMenu()

Comment (hXtreme) [reply app/src/main/java/eu/kanade/tachiyomi/ui/manga/chapter/ChaptersController.kt]: Thanks! I'll take a look at it.

**Comment by joseph619** [comment]
Is this still being worked on or is it ready to merge?

**Comment by Flat** [comment]
It's ready to merge. Could be improved but more effort towards 1.x would be better.

**Comment by hXtreme** [comment]
I'd say it's pretty much ready unless any reviewers want some changes to be made. All the previously requested changes have been implemented.

**Comment by arkon** [comment]
Went with #2087

**Pull request closed without merging**